cpp/models/ode_secirts/parameters_io.h

```diff
@@ -0,0 +1,1158 @@
+ *
+ * @tparam Model The model type used.
+ *
+ * @param[in,out] model Vector of model objects that will be initialized with the input data.
+ * @param[in] date The starting date for the simulation.
+ * @param[in] county Vector of county IDs for which the data is read.
+ * @param[in] scaling_factor_inf Vector of scaling factors for confirmed cases.
+ * @param[in] scaling_factor_icu Scaling factor for ICU data.
+ * @param[in] dir Path to the directory containing input data files.
+ * @param[in] num_days The number of days for which the simulation runs.
+ * @param[in] immunity_population Vector of vectors representing immunity proportions for each age group and immunity layer.
+ * @param[in] export_time_series Boolean flag indicating whether to export time series of extrapolated data (default: false).
+ *
+ * @return An IOResult indicating success or failure.
+ */
+template <class Model>
+IOResult<void> read_input_data_county(std::vector<Model>& model, Date date, const std::vector<int>& county,
+                                      const std::vector<double>& scaling_factor_inf, double scaling_factor_icu,
+                                      const std::string& dir, int num_days,
+                                      const std::vector<std::vector<double>> immunity_population,
+                                      bool export_time_series = false)
+{
+    BOOST_OUTCOME_TRY(details::set_vaccination_data(
+        model, path_join(dir, "pydata/Germany", "all_county_ageinf_vacc_ma7.json"), date, county, num_days));
+
+    // TODO: Reuse more code, e.g., set_divi_data (in secir) and a set_divi_data (here) only need a different ModelType.
+    // TODO: add option to set ICU data from confirmed cases if DIVI or other data is not available.
+    if (date > Date(2020, 4, 23)) {
+        BOOST_OUTCOME_TRY(details::set_divi_data(model, path_join(dir, "pydata/Germany", "county_divi_ma7.json"),
+                                                 county, date, scaling_factor_icu));
+    }
+    else {
+        log_warning("No DIVI data available for this date");
+    }
+
+    BOOST_OUTCOME_TRY(
+        details::set_confirmed_cases_data(model, path_join(dir, "pydata/Germany", "cases_all_county_age_ma7.json"),
+                                          county, date, scaling_factor_inf, immunity_population));
+    BOOST_OUTCOME_TRY(details::set_population_data(
+        model, path_join(dir, "pydata/Germany", "county_current_population.json"), county, immunity_population));
+
+    if (export_time_series) {
+        // Use only if extrapolated real data is needed for comparison. EXPENSIVE !
+        // Run time equals run time of the previous functions times the num_days !
+        // (This only represents the vectorization of the previous function over all simulation days...)
+        log_warning("Exporting time series of extrapolated real data. This may take some minutes. "
+                    "For simulation runs over the same time period, deactivate it.");
```

Comment:
I would say this should be an info, not warning. But you can add the warning that this is expensive to the param documentation. Same for read_input_data below.
```suggestion
        log_info("Exporting time series of extrapolated real data. This may take some minutes. "
                    "For simulation runs over the same time period, deactivate it.");
```